apps/labs/posts/intel-oneapi-msvc-spin-scipy.md

```diff
@@ -0,0 +1,174 @@
+
+Now, let’s talk about the transition from `dev.py` to `spin`.
+`dev.py` (also known as `do.py` [[1]](https://labs.quansight.org/blog/the-evolution-of-the-scipy-developer-cli)) was a significant step forward in improving the development workflow.
+However, `spin` builds on `dev.py`, taking inspiration from its design while generalizing the interface to work with a wider range of projects without the need for heavy customization.
+As a result, SciPy will directly benefit from the improvements made in `spin` and vice versa.
+Moreover, `spin` offers an additional benefit: contributors familiar with `numpy` will find it easier to navigate, thanks to the shared foundation and streamlined approach.
+
+Through these updates, we aim to provide a smoother and more flexible development experience for both current and new contributors to SciPy.
+
+#### Intel oneAPI Support in SciPy
+
+Intel offers oneAPI, a comprehensive toolkit designed for high-performance computing (HPC) applications.
+It includes essential components such as `icx` (for C), `icpx` (for C++), `ifx` (for Fortran), and MKL (Math Kernel Library), which provides optimized mathematical routines like BLAS, LAPACK, and fast FFTs [[2]](https://en.wikipedia.org/wiki/Math_Kernel_Library).
+For our efforts, we targeted both Windows and Linux operating systems.
+
+Our journey began with the task of adding a Windows CI job to support MSVC + MKL + `ifx` [[gh-20878](https://github.com/scipy/scipy/issues/20878)].
+This was critical to ensure that SciPy could compile correctly with Intel oneAPI and MSVC, preventing any regressions.
+This issue also references [gh-20728](https://github.com/scipy/scipy/issues/20728), where the author is building SciPy using `icx`, `icpx`, `ifx`, and MKL, leveraging the full Intel oneAPI toolkit.
+Additionally, the failure to build SciPy with MSVC [[gh-20860]](https://github.com/scipy/scipy/issues/20860) pointed to specific build challenges with MSVC.
+We began by addressing these two issues.
+
+One notable challenge with MSVC is its limited support for handling runtime-sized arrays on the stack.
+For example, expressions like `std::complex<double> cwrk[n];` fail to compile unless `n` is a constant.
+To work around this, we opted for dynamic memory allocation on the heap, using `new`, ensuring proper memory deallocation with `delete` at the end.
+While this approach is manual, it remains the only viable solution with MSVC.
+Interestingly, Clang (due to its LLVM backend) does not have this limitation, making it more flexible in this regard.
+
+Once we addressed this issue, we added a CI job to test SciPy's compilation with MSVC + `ifx` + OpenBLAS, which helped prevent future regressions in the MSVC build.
+This also marked an early step toward supporting Intel oneAPI within SciPy.
+
+The next major hurdle involved the [`arpack`](https://github.com/opencollab/arpack-ng) issue, which required full Intel oneAPI support to resolve.
```

Comment:
I don't think I understand what the "arpack issue" is? Could we link to any issue on the scipy tracker that has more context?

Reply:
I am referring to https://github.com/scipy/scipy/issues/20728. You are right, just linking it here as well should make it clear. I have referred to this issue in lines 42-44.

Reply:
Done. Let me know if it looks good to you.